File: olinfo/gasoline/ois_gasoline/gasoline.cpp
#include <assert.h>
#include <iostream>
#include <vector>
#include <fstream>
using namespace std;

// constraints
#define MAXN 1000000

// input data
int N, i;
vector<long long> P(MAXN);
vector<long long> G(MAXN);

long long gasoline(int N, vector<long long>& P, vector<long long>& G){
    long long result = 0, last_station = 0, passi=0;
    for(int i=0; i<N; i++){
        if(P[i] < P[last_station]){
            result += passi * P[last_station];
            last_station = i;
            passi = G[i];
        }else
            passi += G[i];
    }
    result += passi * P[last_station];
    return result;
}

int main() {
  //  uncomment the following lines if you want to read/write from files
  ifstream in("input.txt");
  ofstream out("output.txt");

  cin >> N;
  for(int i=0; i<N; i++) cin >> P[i];
  for(int i=0; i<N; i++) cin >> G[i];

  // insert your code here
  cout << gasoline(N, P, G);
  return 0;
}

File: olinfo/gasoline/ois_gasoline/gasoline.pas
{ constraints }
const
    MAXN = 1000000;

{ input data }
var
    N, i  : longint;
    P     : array[0..MAXN-1] of longint;
    G     : array[0..MAXN-1] of longint;

begin
{
    uncomment the following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    readln(N);
    for i:=0 to N-1 do
        read(P[i]);
    readln();
    for i:=0 to N-1 do
        read(G[i]);
    readln();

    { insert your code here }

    writeln(42); { print result }
end.
